pkg/controllers/user.go

```diff
@@ -24,23 +23,33 @@ func NewUserController(
 	return &UserController{logger, createUserInteractor}
 }
 
-func (uc *UserController) writeResponse(w http.ResponseWriter, status int, err string) {
-	js, jsonErr := json.Marshal(StatusResponse{Message: err})
-	if jsonErr != nil {
-		uc.logger.Errorf("Unable to marshal json: %s", jsonErr)
+// DetailedErrorResponse represents the response to be sent to the request
```

Comment:
About commit message 4a79d777bc101a2a972e585fb6d208c0fd212fc9:

Please tell in which controller function it belong to.

Reply:
I thought the commit title is already indicating this by using `user` and `ctrl` prefixes. It means "In the context of the user's controller refactor the `writeResponse` method to be generalized".

Do you still think we need more info in the commit message?

Reply:
I think it is enough. I think that when I read the commit on the first time I didn't note the prefixs